libs/community/langchain_community/embeddings/textembed.py

```diff
@@ -0,0 +1,390 @@
+"""
+TextEmbed: Embedding Inference Server
+
+TextEmbed provides a high-throughput, low-latency solution for serving embeddings. 
+It supports various sentence-transformer models. 
+Now, it includes the ability to deploy image embedding models. 
+TextEmbed offers flexibility and scalability for diverse applications.
+
+TextEmbed is maintained by Keval Dekivadiya and is licensed under the Apache-2.0 license.
+"""
+
+import asyncio
+from concurrent.futures import ThreadPoolExecutor
+from typing import Any, Callable, Dict, List, Optional, Tuple
+
+import aiohttp
+import numpy as np
+import requests
+from langchain_core.embeddings import Embeddings
+from langchain_core.pydantic_v1 import BaseModel, Extra, root_validator
+from langchain_core.utils import get_from_dict_or_env
+
+__all__ = ["TextEmbedEmbeddings"]
+
+
+class TextEmbedEmbeddings(BaseModel, Embeddings):
+    """
+    A class to handle embedding requests to the TextEmbed API.
+
+    Attributes:
+        model (str): The TextEmbed model ID to use for embeddings.
+        textembed_api_url (str): The base URL for the TextEmbed API.
+        textembed_api_key (str): The API key for authenticating with the TextEmbed API.
+        client (Any): The TextEmbed client instance.
+
+    Methods:
+        validate_environment(cls, values: Dict) -> Dict:
+            Validates the environment configuration.
+        embed_documents(self, texts: List[str]) -> List[List[float]]:
+            Embeds a list of documents.
+        aembed_documents(self, texts: List[str]) -> List[List[float]]:
+            Asynchronously embeds a list of documents.
+        embed_query(self, text: str) -> List[float]:
+            Embeds a single query.
+        aembed_query(self, text: str) -> List[float]:
+            Asynchronously embeds a single query.
+
+    Example:
+        .. code-block:: python
+
+            from langchain_community.embeddings import TextEmbedEmbeddings
+            embeddings = TextEmbedEmbeddings(
+                model="sentence-transformers/clip-ViT-B-32",
+                textembed_api_url="http://localhost:8000/v1",
+                textembed_api_key="<API_KEY>"
+            )
+    """
+
+    model: str
+    "Underlying TextEmbed model id."
+
+    textembed_api_url: str = "http://localhost:7997"
+    """Endpoint URL to use."""
+
+    textembed_api_key: str = "None"
+    """API Key for authentication"""
+
+    client: Any = None
+    """TextEmbed client."""
+
+    class Config:
+        """Configuration for this pydantic object."""
+
+        extra = Extra.forbid
+
+    @root_validator(allow_reuse=True)
```

Comment:
Please use one of these

```suggestion
    @root_validator(pre=True)
```

OR

```suggestion
    @root_validator(pre=False, skip_on_failure=True)
```

Reply:
resolved